leaderboards/grade_source/mock.py

```diff
@@ -0,0 +1,42 @@
+"""
+Mock code that generates random student grades.
+"""
+import random
+from . import GradeSource
+
+
+class MockGradeSource(GradeSource):
+    """
+    Source of random student grades
+    """
+    ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+    NAMES = ("Jamie", "Alex", "Sam", "Jordan", "Taylor")
+
+    def is_supported(self):
+        """
+        Does self.host_block.runtime support this type of grade source?
+        """
+        return True
+
+    def get_grades(self, target_block_id, limit_hint=None):
+        """
+        Generate random grades for target_block_id and descendants.
+
+        Results should be consistent for use in tests etc.
+        """
+        # Seed a random generator consistent based on target_block_id:
+        seed = 42
+        for letter in unicode(target_block_id):
+            seed += ord(letter)
+        rand = random.Random(seed)
```

Comment:
You should be able to use any hashable object here, according to the docs, even target_block_id itself, which I'd think should be sufficient for this, if you just need consistency with same-ID'd blocks.